MIP/mip-draft-postconfirmation/README.md

```diff
@@ -0,0 +1,238 @@
+
+#### block-range
+
+Validators commit the hash of the block-range to the L1-contract. It commits the validator to the block-range with no option for changing their opinion. (This is intentional - validators should not be able to revert). If an block-range height is new, the L1-contract will asign the block-range height to the `current epoch`. 
+
+```solidity
+function submitBlockCommitmentForAttester(address attester,  BlockCommitment memory blockCommitment) internal {
+  ...
+  if (blockHeightEpochAssignments[blockCommitment.height] == 0) {
+      blockHeightEpochAssignments[blockCommitment.height] = getEpochByL1BlockTime();
+  }
+  ...
+}
+```
+
+!!! TODO leading Block Tolerance
+
+```solidity
+if (lastAcceptedBlockHeight + leadingBlockTolerance < blockCommitment.height) revert AttesterAlreadyCommitted();
+```
+
+> [!note]
+> The validator has to also check if the current relevant block-range height (off-L1) is within the above window otherwise the commitment of the (honest) validator will be not added to the L1 contract.
+
+
+Note that since any validator can commit the hash of a block-range, the height of the block-range should not be able to be set too far into the future.
+
+#### Epochs
+
+We require epochs in order to facilitate `staking` and `unstaking` of validators, as well as rewards and penalties. The `epochDuration` is set when initializing a chain. 
+
+There are two relevant epochs names
+1. The `presentEpoch` is the epoch that is currently active on L1 and it determines the current `acceptor`.
+```solidity
+presentEpoch = getEpochByL1BlockTime();
+```
+where
+```solidity
+function getEpochByL1BlockTime(address domain) public view returns (uint256) {
+    return block.timestamp / epochDuration;
+}
+```
+
+2. The `acceptingEpoch` is the epoch in which commitments are counted and postconfirmation for a block-range is created. If there are no more blocks in the current accepting epoch the protocol progresses from one accepting epoch to the next via the `rollover` function.
+
+> [!note]
+> !!! Should we compensate for the case where validators of an epoch are not active anymore. Should we entertain the possibility that validators of the next epoch can take over to attest for a block from the previous epoch. In principle this may be feasible since the block-range is deterministic, we have L1 as a clock and postconfirmations do not rely on a BFT consensus.
+
+```solidity
+while (getAcceptingEpoch() < blockEpoch) {
+    // TODO: we should check the implication of this for the acceptor. But it also may be an issue for any attester. 
+    rollOverEpoch();
+}
+```
+
+Note that validator L1-transactions could get lost, or validators can become inactive. Since liveness concerns should be handled, we permit for a more recent epoch to accept the block-range from an earlier epoch. 
```

Comment:
This is vague to me. Are you referring to the fact that the rollover can catch up to the actual epoch that would be assigned to the current time?

Reply:
```suggestion
Note that validator L1-transactions could get lost (which means their attestations do not reach L1), or validators can become inactive. Since liveness concerns should be handled, we permit for validators of a more recent epoch to accept the block-range from an earlier epoch. 
```